### PR TITLE
Fix Cache_File TTL=0 to mean "never expires"

### LINKED ISSUE #50: Cache TTL may produce unexpected behaviour, and has a flickering test
As observed in kohana/kohana#41, it seems like one of the inherited test in Kohana_Cache_FileTest may be flickering if the execution runs across two seconds.

The test case in [CacheBasicMethodsTest](https://github.com/kohana/cache/blob/3.4/develop/tests/cache/CacheBasicMethodsTest.php#L106) appears to specify that:
- When ->set is called with a value and TTL 0
- And the test does not wait between set and get
- Then ->get should return the cached value

The [File driver code](https://github.com/kohana/cache/blob/3.4/develop/classes/Kohana/Cache/File.php#L162) checks if a value has expired with the following test:

``` php
if (($created + (int) $lifetime) < time())
```

This usually passes the test, because the < means the value doesn't expire till the end of the second it was set in - but if the test happens to start late in a second so that the get falls into the next second it will fail.

Obviously we need to make the test cope with second boundaries, but I think there's a bigger issue in that I would expect a TTL of 0 to mean "this expires immediately" / "don't cache this at all".

My hunch is that many of the native drivers would fail this test every time. Currently most of their tests are skipped on travis, and APC's TTL support is broken in CLI but I think it would probably also fail in a browser. I need to do a bit more investigation to see if I'm right.

If so, I think this is a bug in 3.3 and we should change the test so that:
- When ->set is called with a value and TTL 0
- And the test does not wait between set and get
- Then ->get should return the default value or NULL

And change the file driver so that it uses `<=` to check for expiry, so items expire at the beginning of the second when their lifetime is up, not the end.

Thoughts?

### PR DESCRIPTION
**Pull request by enov**
The cache expiration test is now negative when
`$lifetime` (TTL) is 0.

Closes #50

### CODE REVIEW

**Comment by acoulton** [comment]
Looks great, thanks @enov.

File: tests/cache/CacheBasicMethodsTest.php [hunk 1]
```diff
@@ -166,6 +166,17 @@ public function provider_set_get()
 			),
 			array(
 				array(
+					'id'      => 'test ttl 0 means never expire',
+					'value'   => 'cache value that should last',
+					'ttl'     => 0,
+					'wait'    => 1,
+					'type'    => 'string',
```
Comment (enov) [inline tests/cache/CacheBasicMethodsTest.php:173]: I feel that this `type` key is misplaced and should go with the `$expected` parameter, because it is used to assert the type in this way:

```
$this->assertInternalType($type, $result);
```

Comment (enov) [inline tests/cache/CacheBasicMethodsTest.php:173]: This is not critical. @acoulton Should I open an issue for this or we just forget about it?

Comment (acoulton) [inline tests/cache/CacheBasicMethodsTest.php:173]: @enov I think you're right - but I don't think it's important enough to have an issue lying around. I'd say fix it now if you have time, or ignore it.

Comment (enov) [inline tests/cache/CacheBasicMethodsTest.php:173]: I'll just ignore it.
